interfaces/system_key.go

```diff
@@ -33,34 +38,68 @@ import (
 type systemKey struct {
 	BuildID          string   `yaml:"build-id"`
 	AppArmorFeatures []string `yaml:"apparmor-features"`
+	NFSHome          bool     `yaml:"nfs-home"`
+	Core             string   `yaml:"core,omitempty"`
 }
 
+var mockedSystemKey *systemKey
+
 func generateSystemKey() *systemKey {
-	var mySystemKey systemKey
+	// for testing only
+	if mockedSystemKey != nil {
+		return mockedSystemKey
+	}
 
-	// add build-id
+	var sk systemKey
 	buildID, err := osutil.MyBuildID()
 	if err != nil {
-		buildID = "unknown"
+		buildID = ""
 	}
-	mySystemKey.BuildID = buildID
+	sk.BuildID = buildID
+
+	// Add apparmor-feature (which is already sorted)
+	sk.AppArmorFeatures = release.AppArmorFeatures()
 
-	// Add apparmor-feature, note that ioutil.ReadDir() is already sorted.
+	// Add if home is using NFS, if so we need to have a different
+	// security profile and if this changes we need to change our
+	// profile.
+	sk.NFSHome, err = osutil.IsHomeUsingNFS()
```

Comment:
That means if we cannot determine nfs usage, we will have sk.NFSHome = false. I'm not sure that's an issue, but perhaps it would make sense to actually use three states: yes, no, unknown? In practical sense it's not relevant for the profiles, but might be handy if we ever have too look at the system key for some other use case.

Reply:
The unknown is not relevant as we only care about weakening the security if the answer was yes. For all practical purposes this behaves as if we did what you said.